ocaml/xapi/xapi_pool.ml

```
@@ -112,6 +112,71 @@ let pre_join_checks ~__context ~rpc ~session_id ~force =
             )
         )
   in
+  let one_ip_configured_on_joining_cluster_network () =
+    match Client.Cluster_host.get_all ~rpc ~session_id with
+    | [] ->
+        ()
+    | ch :: _ -> (
+      match Client.Cluster_host.get_PIF ~rpc ~session_id ~self:ch with
+      | pif when pif = Ref.null ->
+          ()
```

This feels a bit strange, so the existing cluster has a host that does not have a valid PIF? Does that mean the existing cluster is not functioning?

---

Can we do:
```
match Client.Cluster_host.get_PIF ~rpc ~session_id ~self:ch with
| Ref.null ->
| pif ->
```
?

---

Checked related code, yes, now cluster_host will never has a null PIF.
While in case there is such case, since the clustering in the pool is not functioning, we can ignore the error here and just let the host join the pool?

---

```
match Client.Cluster_host.get_PIF ~rpc ~session_id ~self:ch with
| Ref.null ->
| pif ->
```
Can't pass compile with:
`Error: Syntax error`